File: code/taskflow-notification/src/main/java/com/taskflow/notification/entity/Notification.java
package com.taskflow.notification.entity;

import com.taskflow.common.entity.AuditEntity;
import io.hypersistence.utils.hibernate.type.json.JsonBinaryType;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.hibernate.annotations.Type;

import java.util.Map;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@Table(name = "notifications")
public class Notification extends AuditEntity {

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(nullable = false, length = 50)
    private String type;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String body;

    @Column(length = 500)
    private String link;

    @Column(name = "read_at")
    private Long readAt;

    @Type(JsonBinaryType.class)
    @Column(columnDefinition = "jsonb")
    private Map<String, Object> metadata;
}

File: code/taskflow-fe/src/types/notification.ts
export interface Notification {
  id: number;
  user_id: number;
  type: string;
  title: string;
  body?: string | null;
  link?: string | null;
  read_at?: number | null;
  metadata?: Record<string, unknown> | null;
  created_at: number;
}

export interface NotificationPreference {
  user_id: number;
  in_app_enabled: boolean;
  email_enabled: boolean;
  per_type_settings?: Record<string, boolean>;
}

export interface UpdatePreferenceRequest {
  in_app_enabled?: boolean;
  email_enabled?: boolean;
  per_type_settings?: Record<string, boolean>;
}
